File: server/src/main/java/com/fluffy/global/distributedLock/DistributedLock.java
package com.fluffy.global.distributedLock;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;
import java.util.concurrent.TimeUnit;

@Target(ElementType.METHOD)
@Retention(RetentionPolicy.RUNTIME)
public @interface DistributedLock {

    String key();

    TimeUnit timeUnit() default TimeUnit.SECONDS;

    /**
     * 락을 획득하기 위해 대기할 시간
     */
    long waitTime() default 3L;

    /**
     * 락을 유지할 시간
     */
    long leaseTime() default 3L;
}

File: server/src/main/kotlin/com/fluffy/global/distributedLock/DistributedLock.kt
package com.fluffy.global.distributedLock

import java.util.concurrent.TimeUnit

@Target(AnnotationTarget.FUNCTION)
@Retention(AnnotationRetention.RUNTIME)
annotation class DistributedLock(

    val key: String,

    val timeUnit: TimeUnit = TimeUnit.SECONDS,

    /**
     * 락을 획득하기 위해 대기할 시간
     */
    val waitTime: Long = 3L,

    /**
     * 락을 유지할 시간
     */
    val leaseTime: Long = 3L
)
